src/ext/BalExtension/wixext/BalCompiler.cs

```diff
@@ -98,6 +98,37 @@ public override void ParseElement(SourceLineNumberCollection sourceLineNumbers,
         /// <param name="sourceLineNumbers">Source line number for the parent element.</param>
         /// <param name="parentElement">Parent element of element to process.</param>
         /// <param name="attribute">Attribute to process.</param>
+        public override void ParseAttribute(SourceLineNumberCollection sourceLineNumbers, XmlElement parentElement, XmlAttribute attribute)
+        {
+            switch (parentElement.LocalName)
+            {
+                case "BootstrapperApplication":
+                case "BootstrapperApplicationRef":
+                    switch (attribute.LocalName)
+                    {
+                        case "UseCorrectUserCulture":
+                            if (YesNoType.Yes == this.Core.GetAttributeYesNoValue(sourceLineNumbers, attribute))
+                            {
+                                this.Core.CreateVariableRow(sourceLineNumbers, "WixStdBALanguageFix", "1", "numeric", false, false);
```

Comment:
We're trying to avoid introducing new magic variables. See what I did for SuppressDowngradeFailure.